ops/deployer/gauntlet/gauntlet.go

```diff
@@ -0,0 +1,526 @@
+package deployer
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	opsChainlink "github.com/smartcontractkit/chainlink-relay/ops/chainlink"
+	"github.com/smartcontractkit/chainlink-relay/ops/utils"
+	relayUtils "github.com/smartcontractkit/chainlink-relay/ops/utils"
+	"github.com/smartcontractkit/libocr/offchainreporting2/confighelper"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+const (
+	// uploaded code IDs
+	CW20_ID = iota
+	OCR2_ID
+
+	// deployed contract addresses
+	LINK
+	OCR2
+	RequesterAccessController
+	BillingAccessController
+)
+
+type GauntlerDeployer struct {
+	gauntlet relayUtils.Gauntlet
+	network  string
+	Account  map[int]string
+}
+
+func New(ctx *pulumi.Context) (GauntlerDeployer, error) {
+	// check if yarn is installed
+	yarn, err := exec.LookPath("yarn")
+	if err != nil {
+		return GauntlerDeployer{}, errors.New("'yarn' is not installed")
+	}
+	fmt.Printf("yarn is available at %s\n", yarn)
+
+	// Change path to root directory
+	cwd, _ := os.Getwd()
+	os.Chdir(filepath.Join(cwd, "../"))
+
+	fmt.Println("Installing dependencies")
+	if _, err = exec.Command(yarn).Output(); err != nil {
+		return GauntlerDeployer{}, errors.New("error install dependencies")
+	}
+
+	// Generate Gauntlet Binary
+	fmt.Println("Generating Gauntlet binary...")
+	_, err = exec.Command(yarn, "bundle").Output()
+	if err != nil {
+		return GauntlerDeployer{}, errors.New("error generating gauntlet binary")
+	}
+
+	os.Setenv("SKIP_PROMPTS", "true")
+
+	// Check gauntlet works
+	os.Chdir(cwd) // move back into ops folder
+	gauntletBin := filepath.Join(cwd, "../")
+	gauntlet, err := relayUtils.NewGauntlet(gauntletBin)
+
+	if err != nil {
+		return GauntlerDeployer{}, err
+	}
+
+	return GauntlerDeployer{
+		gauntlet: gauntlet,
+		network:  "bombay-testnet",
+		Account:  make(map[int]string),
+	}, nil
+}
+
+func (t *GauntlerDeployer) Load() error {
+	msg := utils.LogStatus("Uploading contract artifacts")
+	err := t.gauntlet.ExecCommand(
+		"upload",
+		t.gauntlet.Flag("network", t.network),
+		"link",
+		"ocr2",
```

Comment:
We need to upload the `access_controller` right?

Reply:
nice catch, yeah we do, I guess the codeIds I already had in there were already valid